yt_dlp/extractor/rai.py

```diff
@@ -182,7 +295,7 @@ def _real_extract(self, url):
         video = media['video']
 
         relinker_info = self._extract_relinker_info(video['content_url'], video_id)
-        self._sort_formats(relinker_info['formats'])
+        self._sort_formats(relinker_info['formats'], ('tbr', 'proto'))
```

Comment:
Explain why these

Reply:
ok, it's not necessary.

while I was testing I saw that sometimes in formats with the same bit-rate hls format had a "priority" over https format and so I enforced this order, but now It seems unnecessary.

Probably because I was still writing the code I was not adding info correctly to the formats list and the sorting was not correct, now it seems to work properly. I'll remove it.

Reply:
Actually, I found a url where this problem exist...

http://www.rai.it/dl/RaiTV/programmi/media/ContentItem-efb17665-691c-45d5-a60c-5301333cbb0c.html


```
λ python -m yt_dlp http://www.rai.it/dl/RaiTV/programmi/media/ContentItem-efb17665-691c-45d5-a60c-5301333cbb0c.html -F[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading webpage
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading video JSON
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading XML metadata for platform mon
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading m3u8 information
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading XML metadata for platform flash
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading XML metadata for platform native
[info] Available formats for efb17665-691c-45d5-a60c-5301333cbb0c:
ID         EXT RESOLUTION FPS |   TBR PROTO  | VCODEC        VBR ACODEC     ABR
---------- --- ---------- --- - ----- ------ - ----------- ----- --------- ----
https-1200 mp4 720x404        | 1266k https  | avc1.4d401f 1266k mp4a.40.2   0k
hls-1266   mp4 720x404    25  | 1266k m3u8_n | avc1.4d401f 1266k mp4a.40.2   0k
```

```
λ python -m yt_dlp http://www.rai.it/dl/RaiTV/programmi/media/ContentItem-efb17665-691c-45d5-a60c-5301333cbb0c.html -F[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading webpage
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading video JSON
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading XML metadata for platform mon
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading m3u8 information
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading XML metadata for platform flash
[Rai] efb17665-691c-45d5-a60c-5301333cbb0c: Downloading XML metadata for platform native
[info] Available formats for efb17665-691c-45d5-a60c-5301333cbb0c:
ID         EXT RESOLUTION FPS |   TBR PROTO  | VCODEC        VBR ACODEC     ABR
---------- --- ---------- --- - ----- ------ - ----------- ----- --------- ----
hls-1266   mp4 720x404    25  | 1266k m3u8_n | avc1.4d401f 1266k mp4a.40.2   0k
https-1200 mp4 720x404        | 1266k https  | avc1.4d401f 1266k mp4a.40.2   0k
```

Reply:
Again, after another test I added the missing FPS tag and now the sorting seems ok.

Reply:
If some formats have fps info and others don't, it will mess up the sorting. In that case, you should use custom sorting. but except for the missing fields, stick close to the default order